common/js_handle.go

```diff
@@ -163,23 +162,7 @@ func (h *BaseJSHandle) GetProperty(_ string) JSHandleAPI {
 }
 
 // JSONValue returns a JSON version of this JS handle.
-func (h *BaseJSHandle) JSONValue() string { // TODO: return error
-	convertToString := func(ro *cdpruntime.RemoteObject) string {
-		res, err := valueFromRemoteObject(h.ctx, ro)
-		if err != nil {
-			k6ext.Panic(h.ctx, "extracting value from remote object: %w", err)
-		}
-		switch v := res.(type) {
-		case string:
-			return v
-		default:
-			buf, err := json.Marshal(v)
-			if err != nil {
-				k6ext.Panic(h.ctx, "marshaling value: %w", err)
-			}
-			return string(buf)
-		}
-	}
+func (h *BaseJSHandle) JSONValue() (string, error) {
 	remoteObject := h.remoteObject
 	if remoteObject.ObjectID != "" {
 		var result *runtime.RemoteObject
```

Comment:
Do we still need `result` here? What about setting the action's result to `remoteObject` directly?

Reply:
Refactored in dc0abe90fb32958389a4e143fbbb725140f60ad1

Reply:
Closing per https://github.com/grafana/xk6-browser/pull/1190/commits/dc0abe90fb32958389a4e143fbbb725140f60ad1.